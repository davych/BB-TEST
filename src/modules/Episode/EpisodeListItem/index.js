import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import withWidth, { isWidthUp } from '@material-ui/core/withWidth'
import Typography from '@material-ui/core/Typography'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import useStyles from './styles'
import Link from '@material-ui/core/Link'

const EpisodeListItem = (props) => {
  const classes = useStyles()
  return (
    <Link className={classes.link} to={`/eposide/${props.data.id}`}>
      <Card className={classes.root}>
        <CardMedia
          className={classes.cover}
          image={
            isWidthUp('sm', props.width)
              ? props.data?.image?.original
              : props.data?.image?.medium
          }
        />
        <div className={classes.details}>
          <CardContent className={classes.content}>
            <Typography
              component="h5"
              variant="h5"
              className={classes.title}
              noWrap
            >
              {props.data.name}
            </Typography>
            <Typography variant="subtitle1" color="textSecondary">
              <div
                className={classes.ellipsisMulti}
                dangerouslySetInnerHTML={{ __html: props.data?.summary }}
              ></div>
            </Typography>
          </CardContent>
        </div>
      </Card>
    </Link>
  )
}

EpisodeListItem.propTypes = {
  data: PropTypes.object,
  width: PropTypes.any,
}

export default withWidth()(EpisodeListItem)
