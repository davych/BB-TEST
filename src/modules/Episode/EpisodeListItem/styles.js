import { makeStyles } from '@material-ui/core/styles';

export default makeStyles((theme) => ({
  root: {
    width: '90%',
    display: 'flex',
    margin: `${theme.spacing(2)}px auto`,
  },
  details: {
    display: 'flex',
    flexDirection: 'column',
  },
  content: {
    flex: '1 0 auto',
  },
  cover: {
    width: 151,
    flex: '1 0 auto',
  },
  title: {
    width: '90%',
  },
  ellipsisMulti: {
    '& > *': {
      overflow: 'hidden',
      'text-overflow': 'ellipsis',
      display: '-webkit-box',
      '-webkit-line-clamp': 3,
      '-webkit-box-orient': 'vertical',
    },
  },
}));
