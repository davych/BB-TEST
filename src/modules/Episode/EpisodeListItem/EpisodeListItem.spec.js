import React from 'react';
import { render, screen } from '@testing-library/react';
import EpisodeListItem from './';
test('renders episode list item with width 599', () => {
  const app = render(<EpisodeListItem data={mockEpisode} width="xs" />);
  expect(app.asFragment()).toMatchSnapshot();
});

// test('renders episode list item with width 601', () => {
//   const app = render(<EpisodeListItem data={mockEpisode} width="md" />);
//   expect(app.asFragment()).toMatchSnapshot();
// });

const mockEpisode = {
  image: {
    original: 'https://static.tvmaze.com/uploads/images/original_untouched/81/202627.jpg',
    medium: 'https://static.tvmaze.com/uploads/images/original_untouched/81/202627.jpg',
  },
  name: 'name',
  summary: '<div>hi</div>hi',
};
