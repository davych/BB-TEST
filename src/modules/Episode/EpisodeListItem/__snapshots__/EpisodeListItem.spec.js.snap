// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders episode list item 1`] = `<DocumentFragment />`;

exports[`renders episode list item with width 599 1`] = `
<DocumentFragment>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-1 MuiTypography-colorPrimary"
    to="/eposide/undefined"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-2 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardMedia-root makeStyles-cover-5"
      />
      <div
        class="makeStyles-details-3"
      >
        <div
          class="MuiCardContent-root makeStyles-content-4"
        >
          <h5
            class="MuiTypography-root makeStyles-title-6 MuiTypography-h5 MuiTypography-noWrap"
          />
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
          >
            <div
              class="makeStyles-ellipsisMulti-7"
            />
          </h6>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;

exports[`renders episode list item with width 601 1`] = `
<DocumentFragment>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-link-8 MuiTypography-colorPrimary"
    to="/eposide/undefined"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-9 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardMedia-root makeStyles-cover-12"
      />
      <div
        class="makeStyles-details-10"
      >
        <div
          class="MuiCardContent-root makeStyles-content-11"
        >
          <h5
            class="MuiTypography-root makeStyles-title-13 MuiTypography-h5 MuiTypography-noWrap"
          />
          <h6
            class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorTextSecondary"
          >
            <div
              class="makeStyles-ellipsisMulti-14"
            />
          </h6>
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
